addons/sale/models/sale.py

```diff
@@ -1368,7 +1368,7 @@ def _compute_untaxed_amount_to_invoice(self):
                     # As included taxes are not excluded from the computed subtotal, `compute_all()` method
                     # has to be called to retrieve the subtotal without them.
                     # `price_reduce_taxexcl` cannot be used as it is computed from `price_subtotal` field. (see upper Note)
-                    price_subtotal = line.tax_id.compute_all(price_subtotal)['total_excluded']
+                    price_subtotal = line.tax_id.compute_all(price_subtotal, line.order_id.currency_id, line.product_uom_qty, product=line.product_id, partner=line.order_id.partner_shipping_id)['total_excluded']
```

Comment:
Just for consistency:
```suggestion
                    price_subtotal = line.tax_id.compute_all(price_subtotal, currency_id=line.order_id.currency_id, quantity=line.product_uom_qty, product=line.product_id, partner=line.order_id.partner_shipping_id)['total_excluded']
```

Reply:
Ok, that was done like that there :-) https://github.com/odoo/odoo/blob/12.0/addons/sale/models/sale.py#L989

Reply:
We have room for improvement in many places :smile:

Reply:
No doubt :smile: